kratos/python_scripts/testing/utilities.py

```diff
@@ -75,3 +78,95 @@ def ReadDistributedModelPart(mdpa_file_name, model_part, importer_settings):
         model_part, importer_settings)
     model_part_import_util.ImportModelPart()
     model_part_import_util.CreateCommunicators()
+
+
+def PrintTestHeader(application):
+    Kratos.Logger.Flush()
+    print("\nRunning {} tests".format(application), file=sys.stderr, flush=True)
+
+def PrintTestFooter(application, exit_code):
+    Kratos.Logger.Flush()
+    appendix = " with exit code {}!".format(exit_code) if exit_code != 0 else "."
+    print("Completed {} tests{}\n".format(application, appendix), file=sys.stderr, flush=True)
+
+def PrintTestSummary(exit_codes):
+    Kratos.Logger.Flush()
+    print("Test results summary:", file=sys.stderr, flush=True)
+    max_test_name_length = len(max(exit_codes.keys(), key=len))+1
+    for test, exit_code in exit_codes.items():
+        result_string = "OK" if exit_code == 0 else "FAILED"
+        pretty_name = test.ljust(max_test_name_length)
+        print("  {}: {}".format(pretty_name, result_string), file=sys.stderr, flush=True)
+    sys.stderr.flush()
+
+class Commander:
+    def __init__(self):
+        self.process = None
+        self.exitCode = 0
+
+    def RunMPITestSuit(self, application, path, num_processes, level, verbose, command, timer):
+        ''' Calls the script that will run the tests.
+
+        Input
+        -----
+        application: string
+            Name of the application that will be tested.
+
+        path: string
+            Absoulte path with the location of the application.
+
+        level: string
+            minimum level of the test that will be run if possible.
+
+        verbose: int
+            detail of the ouptut. The grater the verbosity level, the greater the detail will be.
+
+        command: string
+            command to be used to call the tests. Ex: Python, Python3, Runkratos
+
+        '''
+
+        if not level.startswith("mpi_"):
+            raise Exception("Trying to run a serial suite!")
+
+        self.exitCode = 0
+
+        test_script = path / Path("tests") / Path("test_{}.py".format(application))
+
+        if Path.is_file(test_script):
+            full_command = "mpiexec --oversubscribe -np {} {} {} --using-mpi".format(num_processes, command, test_script)
```

Comment:
I would allow the user to override the mpi runner and the np flag if possible, sometimes `mpiexec` is not available directly or you are required to run through other scrips with different syntaxes like:

* `aprun -N 4`
* `srun -n 4`
* ...etc

Something like:

```
full_command = "{} {} {} {} {} {} --using-mpi".format(mpi_command, mpi_flags, num_processes_flag, num_processes, command, test_script)
```

Leaving the ones in there as the default ones.

Reply:
done